extendr-api/src/lib.rs

```diff
@@ -370,6 +370,36 @@ impl IsNA for &str {
     }
 }
 
+/// Type of R objects used by [Robj::rtype].
+#[derive(Debug, PartialEq)]
+pub enum RType {
+    Null,
+    Symbol,
+    Function,
+    Enviroment,
+    Pairlist,
+    Promise,
+    Language,
+    Special,
+    Builtin,
+    Character,
+    Logical,
+    Integer,
+    Real,
+    Complex,
+    String,
+    Dot,
+    Any,
+    List,
+    Expression,
+    Bytecode,
+    ExternalPtr,
+    WeakRef,
+    Raw,
+    S4,
+    Unknown,
```

Comment:
Could you add comments here for each stating which SEXPTYPE it corresponds to? Like so?

```
Logical,  // LGLSXP
Integer,  // INTSXP
Real,       // REALSXP
```
etc?